docs/features/logforwarding/custom-tenants/custom-tenant-support.adoc

```diff
@@ -0,0 +1,246 @@
+
+== Elasticsearch
+Elasticsearch's default behavior is to sends logs to indices defined by their `log_type` and appended by `-write`.
+
+  Example: `application` logs will be sent to the index `app-write`.
+
+ElasticSearch's index can be configured if `parse: json` is enabled.
+
+* Configure using the `structuredTypeKey & structuredTypeName` fields.
+
+=== Example
+
+.cluster-log-forwarder.yaml
+[source,yaml]
+----
+spec:
+  outputs:
+    - name: external-es
+      type: elasticsearch
+      elasticsearch: 
+        structuredTypeKey: kubernetes.namespace_name <1>
+        structuredTypeName: myParsedMessages <2>
+  pipelines:
+    - name: parsed-app-logs
+      inputRefs:
+        - application
+      outputRefs:
+        - external-es
+      parse: json <3>
+
+----
+<1> Specify the `structuredTypeKey` here.
+<2> Specify the `structuredTypeName` here.
+<3> Parse JSON must be enabled.
+
+NOTE: Both `structuredTypeKey` & `structuredTypeName` must be specified.
+
+=== References
+
+. See the link:../outputs/elasticsearch-forwarding.adoc[custom index configuration section] of the internal Elasticsearch forwarding documentation.
+
+. See also the link:https://docs.openshift.com/container-platform/4.14/logging/log_collection_forwarding/cluster-logging-enabling-json-logging.html#cluster-logging-configuration-of-json-log-data-for-default-elasticsearch_cluster-logging-enabling-json-logging[configuring JSON log data for Elasticsearch] on the offical documention.
+
+== Google Cloud Logging
+Google Cloud Logging can be configured with a custom tenant by setting the `logId` field.
+
+=== Example
+
+.cluster-log-forwarder.yaml
+[source,yaml]
+----
+spec:
+  outputs:
+    - name: gcp-1
+      type: googleCloudLogging
+      secret:
+        name: gcp-secret
+      googleCloudLogging:
+        projectId : openshift-gce-devel
+        logId : app-gcp <1>
+----
+<1> Specify the `logId` here.
+
+=== References
+
+. See the link:../outputs/google-cloud-forwarding.adoc[internal forwarding doc] for Google Cloud Logging.
+
+. See also the link:https://docs.openshift.com/container-platform/4.14/logging/log_collection_forwarding/configuring-log-forwarding.html#cluster-logging-collector-log-forward-gcp_configuring-log-forwarding[forwarding logs to GCL section] on the official documentation.
+
+== HTTP
+A `header` can be added to the HTTP output as a tenant label.
+
+=== Example
+
+.cluster-log-forwarder.yaml
+[source,yaml]
+----
+spec:
+  outputs:
+    - name: httpout-app
+      type: http
+      url: <HTTP-URL>
+      http:
+        headers: 
+          h1: v1
+          tenant: myAppLogs <1>
+        method: POST
+----
+<1> Specify the `header` here.
+
+=== References
+
+. See the link:../outputs/send-logs-to-fluentd-http.adoc[internal forwarding doc] for fluentd-http.
```

Comment:
The mention of fluentd-http was confusing to me.   Do we want to instead mention that this is a 'vector' feature.   So maybe something like "example vector http"   (Do we need to mention fluentd, they'll figure it out once they click through??).